src/main/java/com/okta/tools/aws/settings/Settings.java

```diff
@@ -15,22 +15,26 @@
  */
 package com.okta.tools.aws.settings;
 
-import org.ini4j.Config;
-import org.ini4j.Ini;
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.commons.configuration2.ex.ConfigurationException;
+import org.apache.commons.configuration2.tree.ImmutableNode;
 
 import java.io.IOException;
 import java.io.Reader;
 import java.io.Writer;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
 
 /**
  * A simple abstract class used to load AWS CLI settings files like the `credentials` or `config` files.
  */
 public abstract class Settings {
 
-    final Ini settings = new Ini();
+    private final AwsINIConfiguration settings;
```

Comment:
A huge chunk of the change stems from hiding this as exposing it causing Ini4j's API to leak into everywhere and my original attempt at this had Apache Commons Configuration2's API leaked into everywhere.